File: class/l4/sumpair.rb
#!/usr/bin/env ruby

a = [1,2,3,4,5]
b = [6,3,5,1,2]
x = 10

#debug: print the input
print "a = "
a.each { |i| print "#{i} " }
print "\nb = "
b.each { |i| print "#{i} " }
puts 

#instruction count
count = 0;

#we want to find i,j such that a[i] + b[j] = x
puts "Solutions:"
a.length.times do |i|
  b.length.times do |j|
    count = count + 1
    puts "a[#{i}] + b[#{j}] = #{x}" if a[i] + b[j] == x
  end
end


puts "Count = #{count}"

File: class/l4/sumpair.cpp
#include <stdio.h>
#include <vector>
#include <iostream>

using namespace std;

int main() {
  vector<int> a = {1,2,3,4,5};
  vector<int> b = {6,3,5,1,2};
  int x = 7;

  //this is just a debug
  cout << "a = ";
  for (auto it = a.begin();it != a.end();it++) 
    cout << *it << " ";
  cout << endl << "b = ";
  for (auto it = b.begin();it != b.end();it++) 
    cout << *it << " ";
  cout << endl;


  //we want to find i,j such that a[i] + b[j] = x
  for (size_t i = 0;i < a.size();i++) {
    for (size_t j = 0;j < a.size();j++) {
      if (a[i] + b[j] == x) {
        cout << "a[" << i << "] + b[" << j << "] = " << x << endl;
      }
    }
  }
}
